### ISSUE TITLE
System Memory in Configuration

### ISSUE THREAD

**Issue by aachaemenes** [trimble-oss/dba-dash#1131]
Currently DBA dash under configuration for all the servers shows SQL server settings for MAX and MIN memory. The system memory information is captures under hardware info which is great but If I want to see a quick report on how much ram has been assigned to Windows I have to go back and forth between the tabs. It would fantastic if we can see that info there also. Thanks.

**Comment by aachaemenes** [comment]
Yes we do but the problem is I have to do back and forth. I have 400
servers. Want to able to see how much ram windows has. We found servers
that only had 2 gb ram left and was causing issues.

**Comment by EsQueEl-Fella** [comment]
I suggest you fix the max amount of the RAM for SQL Server to avoid problems such as the one you mentioned above. What is left over is for Windows (which can get along very well with min 4 GB RAM)

**Comment by aachaemenes** [comment]
I agree but seeing them in the same report makes sense. That's all I am
saying.

**Comment by DavidWiseman** [comment]
It's not clear exactly what you are asking for but it sounds like you want to know what the current amount of free memory the OS has & identify current issues?  
The best way to identify where you have current memory issues across servers - regardless of the max memory configuration, is the performance summary tab.  You can add the following performance counters to this summary:
Available Physical Memory (KB) - Current and Min are probably the best aggregations
System Low Memory Signal State - Current and Max are probably the best aggregations

The low signal state should never be 1 - this indicates that the OS is under memory pressure.  If this is 1, you might need to reduce max server memory or address whatever is consuming memory on the server if it's not the SQL engine.

The Hardware tab is useful to check the configuration settings are generally appropriate - leaving a percentage of memory free for the OS.  As a general rule 10%/4GB for the O/S whichever is greater.  If you don't have a dedicated server for 1 instance of the SQL engine, you might need to leave a lot more.  I don't think the current available physical memory is really needed on the hardware tab.

**Comment by aachaemenes** [comment]
Sorry for the confusion. Let say you have 400 servers. And you want to know
right of the bat which one by mistake has too much memory assigned  to SQL
server and windows is almost dying because it has 2 GB ram. How can I see
that quickly in DBA dash in one screen?

**Comment by DavidWiseman** [comment]
As above, you can check the Performance Summary tab.  Add these counters in if needed:

Available Physical Memory (KB) - Current and Min are probably the best aggregations
System Low Memory Signal State - Current and Max are probably the best aggregations

You can save the view so they are always displayed & set thresholds on the counters.  This identifies servers with issues where you might need to either adjust the max server memory or look at what else is consuming memory on the box.

**Comment by aachaemenes** [comment]
Thanks.
